src/scenes/home/Home.js

```diff
@@ -85,6 +86,17 @@ export class Home extends Component {
       task: t,
     });
   }
+  
+  clearAllDone = () => {
+    const { tasks } = this.state;
+    let data = tasks.slice();
+    console.log(data);
+    data = data.filter(x => !x.done);
+    console.log(data);
+    this.setState({
+      tasks: data
+    })
+  }
```

Comment:
You can make this function less verbose refactoring it in this way:
```
clearAllDone = () => {
    this.setState({
      tasks: this.state.tasks
      .filter(x => !x.done)
    })
  }
```
Keep in mind the `filter` does not mutate the state, so there is actually no need if applying a `slice` 🙂